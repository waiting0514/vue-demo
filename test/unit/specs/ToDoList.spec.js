import Vue from 'vue'
import { shallow } from 'vue-test-utils'
import ToDoList from '@/components/ToDoList'

describe('ToDoList.vue', () => {
    let wrapper, vm
    beforeEach(() => {
        wrapper = shallow(ToDoList,{
            data:{
                list:[
                    {
                        checked: true,
                        isEdit: false,
                        content: 'aaa'
                    },
                    {
                        checked: false,
                        isEdit: false,
                        content: 'bbb'
                    },
                    {
                        checked: false,
                        isEdit: false,
                        content: 'ccc'
                    },
                ]
            }
        })
        vm = wrapper.vm
      })
  it('test source list', () => {
    expect(vm.list)
      .toEqual([
            {
                checked: true,
                isEdit: false,
                content: 'aaa'
            },
            {
                checked: false,
                isEdit: false,
                content: 'bbb'
            },
            {
                checked: false,
                isEdit: false,
                content: 'ccc'
            },
        ])
  })
  it('has the expected html structure', () => {
    expect(vm.$el).toMatchSnapshot()
  })
  describe('Event', () => {
    it('新增資料', () => {
        const stub = jest.fn()
        wrapper.setData({newContent:'ddd'})
        wrapper.vm.add()

        expect(vm.list)
        .toEqual([
                {
                    checked: true,
                    isEdit: false,
                    content: 'aaa'
                },
                {
                    checked: false,
                    isEdit: false,
                    content: 'bbb'
                },
                {
                    checked: false,
                    isEdit: false,
                    content: 'ccc'
                },
                {
                    checked: false,
                    isEdit: false,
                    content: 'ddd'
                },
            ])
        })

    it('雙擊第一筆', () => {
        const stub = jest.fn()
        wrapper.vm.edit(0)

        expect(vm.list)
        .toEqual([
                {
                    checked: true,
                    isEdit: true,
                    content: 'aaa'
                },
                {
                    checked: false,
                    isEdit: false,
                    content: 'bbb'
                },
                {
                    checked: false,
                    isEdit: false,
                    content: 'ccc'
                },
            ])
        })

    it('關閉編輯第一筆', () => {
        const stub = jest.fn()
        wrapper.vm.endEdit(0)

        expect(vm.list)
        .toEqual([
                {
                    checked: true,
                    isEdit: false,
                    content: 'aaa'
                },
                {
                    checked: false,
                    isEdit: false,
                    content: 'bbb'
                },
                {
                    checked: false,
                    isEdit: false,
                    content: 'ccc'
                },
            ])
        })

    it('刪除第一筆', () => {
        const stub = jest.fn()
        wrapper.vm.del(0)

        expect(vm.list)
        .toEqual([
              {
                  checked: false,
                  isEdit: false,
                  content: 'bbb'
              },
              {
                  checked: false,
                  isEdit: false,
                  content: 'ccc'
              },
          ])
      })

      it('點擊顯示所有', () => {
        const stub = jest.fn()
        wrapper.vm.filterAll()

        expect(vm.filter)
        .toEqual(0)
      })
      it('點擊顯示未完成', () => {
        const stub = jest.fn()
        wrapper.vm.filterActive()

        expect(vm.filter)
        .toEqual(1)
      })  
      it('點擊顯示已完成', () => {
        const stub = jest.fn()
        wrapper.vm.filterCompleted()

        expect(vm.filter)
        .toEqual(2)
      })  
  })
})
