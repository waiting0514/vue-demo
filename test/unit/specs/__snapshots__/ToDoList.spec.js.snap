// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDoList.vue has the expected html structure 1`] = `
<section
  class="todoapp"
>
  <header
    class="header"
  >
    <h1>
      todos
    </h1>
     
    <input
      autofocus="autofocus"
      class="new-todo"
      placeholder="What needs to be done?"
    />
  </header>
   
  <section
    class="main"
  >
    <input
      class="toggle-all"
      id="toggle-all"
      type="checkbox"
    />
     
    <label
      for="toggle-all"
    >
      Mark all as complete
    </label>
     
    <ul
      class="todo-list"
    >
      <li
        class="completed"
      >
        <div
          class="view"
        >
          <input
            class="toggle"
            type="checkbox"
          />
           
          <label>
            aaa
          </label>
           
          <button
            class="destroy"
          />
        </div>
         
        <input
          class="edit"
        />
      </li>
      <li
        class=""
      >
        <div
          class="view"
        >
          <input
            class="toggle"
            type="checkbox"
          />
           
          <label>
            bbb
          </label>
           
          <button
            class="destroy"
          />
        </div>
         
        <input
          class="edit"
        />
      </li>
      <li
        class=""
      >
        <div
          class="view"
        >
          <input
            class="toggle"
            type="checkbox"
          />
           
          <label>
            ccc
          </label>
           
          <button
            class="destroy"
          />
        </div>
         
        <input
          class="edit"
        />
      </li>
    </ul>
  </section>
   
  <footer
    class="footer"
  >
    <span
      class="todo-count"
    >
      <strong>
        2
      </strong>
       item left
    </span>
     
    <ul
      class="filters"
    >
      <li>
        <a
          class="selected"
        >
          All
        </a>
      </li>
       
      <li>
        <a
          class=""
        >
          Active
        </a>
      </li>
       
      <li>
        <a
          class=""
        >
          Completed
        </a>
      </li>
    </ul>
     
    <button
      class="clear-completed"
    >
      Clear completed
    </button>
  </footer>
</section>
`;
